miner/worker.go

```diff
@@ -775,11 +812,204 @@ func (w *worker) updateSnapshot() {
+				}
+				stgProof = types.WrapProof(proof)
+				proofCache.storageProof[keyStr] = stgProof
+			}
+			// isDelete
+			if proofCache.trieTracer.Available() && bytes.Equal(values.Bytes(), common.Hash{}.Bytes()) {
+				proofCache.trieTracer.MarkDeletion(key)
+			}
+			txStorageTrace.StorageProofs[addrStr][keyStr] = stgProof
+		}
+
+		// build dummy per-tx deletion proof
+		if proofCache.trieTracer.Available() {
+			delProofs, err := proofCache.trieTracer.GetDeletionProofs()
+			if err != nil {
+				log.Error("deletion proof failure", "error", err)
+			} else {
+				for _, proof := range delProofs {
+					txStorageTrace.DeletionProofs = append(txStorageTrace.DeletionProofs, proof)
+				}
+			}
+		}
+	}
+
+	createdAcc := tracer.CreatedAccount()
+	if to == nil {
+		if createdAcc == nil {
+			return nil, errors.New("unexpected tx: address for created contract unavailable")
+		}
+		to = &createdAcc.Address
+	}
+	var after []*types.AccountWrapper
+	// collect affected account after tx being applied
+	for _, acc := range []common.Address{from, *to, coinbase} {
+		after = append(after, &types.AccountWrapper{
+			Address:          acc,
+			Nonce:            w.current.state.GetNonce(acc),
+			Balance:          (*hexutil.Big)(w.current.state.GetBalance(acc)),
+			KeccakCodeHash:   w.current.state.GetKeccakCodeHash(acc),
+			PoseidonCodeHash: w.current.state.GetPoseidonCodeHash(acc),
+			CodeSize:         w.current.state.GetCodeSize(acc),
+		})
+	}
+
+	traces := &types.BlockTrace{
+		ChainID: w.chainConfig.ChainID.Uint64(),
+		Version: params.ArchiveVersion(params.CommitHash),
+		Header:  w.current.header,
+		Coinbase: &types.AccountWrapper{
+			Address:          coinbase,
+			Nonce:            w.current.state.GetNonce(coinbase),
+			Balance:          (*hexutil.Big)(w.current.state.GetBalance(coinbase)),
+			KeccakCodeHash:   w.current.state.GetKeccakCodeHash(coinbase),
+			PoseidonCodeHash: w.current.state.GetPoseidonCodeHash(coinbase),
+			CodeSize:         w.current.state.GetCodeSize(coinbase),
+		},
+		WithdrawTrieRoot: withdrawtrie.ReadWTRSlot(rcfg.L2MessageQueueAddress, w.current.state),
+		Transactions: []*types.TransactionData{
+			types.NewTransactionData(tx, w.current.header.Number.Uint64(), w.chainConfig),
+		},
+		ExecutionResults: []*types.ExecutionResult{
+			{
+				From:           sender,
+				To:             receiver,
+				AccountCreated: createdAcc,
+				AccountsAfter:  after,
+				Gas:            receipt.GasUsed,
+				Failed:         receipt.Status == types.ReceiptStatusFailed,
+				ReturnValue:    fmt.Sprintf("%x", common.CopyBytes(receipt.ReturnValue)),
+				StructLogs:     vm.FormatLogs(tracer.StructLogs()),
+			},
+		},
+		StorageTrace:   txStorageTrace,
+		TxStorageTrace: []*types.StorageTrace{txStorageTrace},
+	}
+
+	if receipt.L1Fee != nil {
+		traces.ExecutionResults[0].L1Fee = receipt.L1Fee.Uint64()
+	}
+
+	// probably a Contract Call
+	if len(tx.Data()) != 0 && tx.To() != nil {
```

Comment:
Even if the tx data is empty, it's probably a contract call of the fallback function.